src/hotspot/share/code/codeCache.hpp

```diff
@@ -111,8 +112,12 @@ class CodeCache : AllStatic {
 
   // CodeHeap management
   static void initialize_heaps();                             // Initializes the CodeHeaps
-  // Check the code heap sizes set by the user via command line
-  static void check_heap_sizes(size_t non_nmethod_size, size_t profiled_size, size_t non_profiled_size, size_t cache_size, bool all_set);
+  static void report_cache_size_error(const CodeCacheSegment& non_nmethod, const CodeCacheSegment& profiled, const CodeCacheSegment& non_profiled, size_t cache_size);
+  static void report_cache_minimal_size_error(const char *codeheap, size_t size, size_t required_size);
+  static size_t subtract_size(size_t cache_size, size_t known_segments_size, size_t min_size) {
+    return (cache_size > known_segments_size + min_size) ? (cache_size - known_segments_size) : min_size;
+  }
+
```

Comment:
Do we need them in `CodeCache` class and in the hpp file?
Why not to have them as static functions in the cpp file? In such a case there will to expose the `CodeCacheSegment` struct.

Reply:
OK